### PR TITLE
Add changelog

### PR DESCRIPTION
**Pull request by logan-markewich**
# Description

We desperately need a changelog. This PR adds that, as well as a workflow check so that every PR should (hopefully) be added to the changelog.

I know it seems annoying, but it's a small thing that will GREATLY improve the usability of the repo.

Future work is adding a hook to display the changelog in our docs.

Not 100% if the workflow check will work yet though.

Fixes https://github.com/jerryjliu/llama_index/issues/6184

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -27,4 +29,19 @@ jobs:
...
+        run: |
+          if ${{ github.event_name == 'pull_request' }}; then
+            echo "changed_files=$(git diff --name-only -r HEAD^1 HEAD | xargs)" >> $GITHUB_OUTPUT
+          else
+            echo "changed_files=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }} | xargs)" >> $GITHUB_OUTPUT
+          fi
+      - name: Ensure Changelog Updated
+        run: |
+          for file in ${{ steps.changed-files.outputs.changed_files }}; do 
+            if [ "${file}" == CHANGELOG.md ]; then
```
Comment (Disiok) [inline .github/workflows/lint.yml:43]: I wonder if this is too strict. We currently do some hotfix PRs that fixes a couple of lines.

(I think in principle we shouldn't do that.) But not sure if the added friction here is necessary.

**Review by Disiok — APPROVED** [review]
Thanks for looking into this. Gentle pushback on enforcing always adding changelog (see comment).

File: .github/workflows/lint.yml (hunk 1 above)
Comment (Disiok) [reply .github/workflows/lint.yml:43]: I think the concern here is that we'd end up with placeholder changelog lines if we really enforce every PR modifies it.

Comment (logan-markewich) [reply .github/workflows/lint.yml:43]: That's fair. I can remove the check, but as we just need to do a good job of policing the file at our discretion 👍

Maybe I'll add a note to to the PR template instead

**Pull request merged**